### PR TITLE
Make `server.socket.*` attributes on the HTTP server side opt-in

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
This is just the span attributes; for metric attributes I'm hoping that [my idea to use the advice API](https://github.com/open-telemetry/opentelemetry-specification/pull/3546#issuecomment-1594367578) catches on.

### CODE REVIEW

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpServerAttributesExtractorBuilder.java [hunk 1]
```diff
@@ -64,12 +65,30 @@ public HttpServerAttributesExtractorBuilder<REQUEST, RESPONSE> setCapturedRespon
...
+   *
+   * @param captureServerSocketAttributes {@code true} if the extractor should collect the optional
+   *     {@code server.socket.address} and {@code server.socket.port} attributes.
+   */
+  @CanIgnoreReturnValue
+  public HttpServerAttributesExtractorBuilder<REQUEST, RESPONSE> setCaptureServerSocketAttributes(
+      boolean captureServerSocketAttributes) {
+    this.captureServerSocketAttributes = captureServerSocketAttributes;
+    return this;
+  }
```
Comment (trask) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpServerAttributesExtractorBuilder.java:80]: My understanding is that we don't have to support opt-in things. I'd personally suggest we not provide this configuration unless someone has specific need now, and we can wait to see how your advice PR pans out.

Comment (mateuszrzeszutek) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpServerAttributesExtractorBuilder.java:80]: [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/3972c6aafd8f14e305dbbb0b5b6dded9d02d6164/specification/common/attribute-requirement-level.md#opt-in)

> Instrumentations SHOULD populate the attribute if and only if the user configures the instrumentation to do so. Instrumentation that doesn't support configuration MUST NOT populate Opt-In attributes.

It's not very explicit about optional support for these; although there is a "SHOULD" here, so it flies, I think. I'm fine with that; and also this is a good subject for the general advice proposed in #3557

**Comment by mateuszrzeszutek** [comment]
@open-telemetry/java-instrumentation-approvers please review this PR 🙏

**Review by laurit — APPROVED** [review]

**Comment by trask** [comment]
@lmolkova do you remember why `server.socket.*` are opt-in for HTTP server spans?

**Comment by lmolkova** [comment]
> @lmolkova do you remember why `server.socket.*` are opt-in for HTTP server spans?

One reason I can think about is that they represent local host/port and are the same for all spans coming from a typical single-host application - this way they're captured best with resource attributes

**Pull request merged**